### PR TITLE
Dial back aggressive checks in Path

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Aggressive checks are preventing crossplat and extended Windows solutions. This change:

- Doesn't throw on empty paths fro GetDirectoryName, GetPathRoot
- Doesn't consider colon when looking at path segments on Windows
- Moves non-shared code out of PathInternal
- Fix span GetDirectoryName to handle multiple separators

I've marked this as no merge- I'll be linking in the necessary test fixes (not many).

### CODE REVIEW

File: src/mscorlib/shared/System/IO/Path.Unix.cs [hunk 1]
```diff
@@ -120,11 +120,9 @@ public static bool IsPathRooted(ReadOnlySpan<char> path)
         // only contains whitespace characters an ArgumentException gets thrown.
         public static string GetPathRoot(string path)
```
Comment (danmosemsft) [inline src/mscorlib/shared/System/IO/Path.Unix.cs]: comment above is now wrong.

Comment (JeremyKuhne) [reply src/mscorlib/shared/System/IO/Path.Unix.cs]: Thanks.

**Comment by JeremyKuhne** [comment]
Test updates were merged: https://github.com/dotnet/corefx/pull/27270

**Comment by JeremyKuhne** [comment]
@dotnet-bot test Tizen armel Cross Checked Innerloop Build and Test 

Timed out

**Comment by jkotas** [comment]
@Anipik The mirror skipped this PR, and also https://github.com/dotnet/corefx/pull/27328 . Could you please take a look ?

**Comment by Anipik** [comment]
@jkotas  I will add the commit after https://github.com/dotnet/corefx/pull/27381 has merged

**Comment by ahsonkhan** [comment]
> I will add the commit after https://github.com/dotnet/corefx/pull/27381 has merged

It has been merged.

**Comment by jkotas** [comment]
@Anipik This one was still not mirrored yet. Could you please take a look?

**Pull request merged**